internal/controller/datadogagent/finalizer.go

```diff
@@ -93,6 +94,23 @@ func (r *Reconciler) finalizeDadV2(reqLogger logr.Logger, obj client.Object) err
 
 	var errs []error
 
+	// Global dependencies
```

Comment:
We can split out dependency handling into a separate function in a future pr to deduplicate code here